src/sparseml/pytorch/utils/sparsification_info/configs.py

```diff
@@ -12,20 +12,175 @@
+                count=num_quantized_ops, percent=num_quantized_ops / len(operations)
+            ),
+            parameter_counts=count_parameters,
+            operation_counts=Counter([op.__class__.__name__ for op in operations]),
+        )
 
 
 class SparsificationPruning(BaseModel):
-    pass
+    """
+    A model that contains the pruning information for a torch module.
+    """
+
+    zero_parameters: Dict[str, CountAndPercent] = Field(
+        description="A dictionary that maps the name of a parameter "
+        "to the number/percent of weights that are zeroed out "
+        "in that layer."
+    )
+
+    @classmethod
+    def from_module(cls, module: torch.nn.Module) -> "SparsificationPruning":
+        """
+        Factory method to create a SparsificationPruning object from a module.
+
+        :param module: The module to create the SparsificationPruning object from.
+        :return: A SparsificationPruning object.
+        """
+        zero_parameters_count = defaultdict(CountAndPercent)
+        for param_name, param in module.named_parameters():
+            num_parameters = param.numel()
+            num_zero_parameters = param.numel() - param.count_nonzero().item()
+
+            zero_count = num_zero_parameters
+            zero_count_percent = num_zero_parameters / num_parameters
+
+            zero_parameters_count[param_name] = CountAndPercent(
+                count=zero_count, percent=zero_count_percent
+            )
+
+        return cls(zero_parameters=zero_parameters_count)
 
 
 class SparsificationQuantization(BaseModel):
-    pass
+    """
+    A model that contains the quantization information for a torch module.
+    """
+
+    enabled: Dict[str, bool] = Field(
+        description="A dictionary that maps the name of an "
+        "operation to a boolean flag that indicates whether "
+        "the operation is quantized or not."
+    )
+    quantization_schema: Dict[str, Union[BaseModel, None]] = Field(
+        description="A dictionary that maps the name of a layer"
+        "to the dtype (precision) of that layer."
+    )
+
+    @classmethod
+    def from_module(
+        cls,
+        module: torch.nn.Module,
+    ) -> "SparsificationQuantization":
+        """
+        Factory method to create a SparsificationQuantization object from a module.
+
+        :param module: The module to create the SparsificationQuantization object from.
+        :return: A SparsificationQuantization object.
+        """
+        operations = get_leaf_operations(module)
+        enabled = defaultdict(bool)
+        quantization_schema = defaultdict(str)
+        for op in operations:
+            operation_name = op.__class__.__name__
+            operation_counter = 0
+            # make sure that the operation name is unique
+            while enabled.get(operation_name) is not None:
+                operation_counter += 1
+                operation_name = f"{op.__class__.__name__}_{operation_counter}"
```

Comment:
Could just move operation_counter out of the loop, but you will get some operations that could start at 0 that don't
```suggestion
        quantization_schema = defaultdict(str)
        operation_counter = 0
        for op in operations:
            operation_name = f"{op.__class__.__name__}_{operation_counter}"
            operation_counter += 1
```